platform/security/core/security-core-impl/src/main/java/ddf/security/assertion/impl/SecurityAssertionImpl.java

```diff
@@ -337,6 +369,45 @@ public SecurityToken getSecurityToken() {
         return securityToken;
     }
 
+    /**
+     * Checks if the NameIDFormat is of the following formats below, if not, the name is changed
+     * to the value of the first matching usernameAttribute.
+     */
+    public void identifyNameIDFormat() {
+        if ((StringUtils.containsIgnoreCase(nameIDFormat, SAML2Constants.NAMEID_FORMAT_PERSISTENT)
+                || StringUtils
+                .containsIgnoreCase(nameIDFormat, SAML2Constants.NAMEID_FORMAT_X509_SUBJECT_NAME) ||
+                StringUtils.containsIgnoreCase(nameIDFormat, SAML2Constants.NAMEID_FORMAT_KERBEROS)
+                || StringUtils
+                .containsIgnoreCase(nameIDFormat, SAML2Constants.NAMEID_FORMAT_UNSPECIFIED))
+                && !name.equals("")) {
+            return;
+        } else {
+            for (AttributeStatement attributeStatementList : getAttributeStatements()) {
+                List<Attribute> attributeList = attributeStatementList.getAttributes();
+                for (Attribute attribute : attributeList) {
+                    if (listContainsIgnoreCase(usernameAttributeList, attribute.getName())) {
+                        List<XMLObject> attributeValues = attribute.getAttributeValues();
```

Comment:
Why did you loop through the values here just to grab the first one and return? Just grab the first one explicitly.